sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,422 @@
+                random_state=None,
+            ):
+                super().__init__(
+                    alpha=alpha,
+                    fit_intercept=fit_intercept,
+                    copy_X=copy_X,
+                    max_iter=max_iter,
+                    tol=tol,
+                    solver=solver,
+                    positive=positive,
+                    random_state=random_state,
+                )
+
+        else:
+
+            def __init__(
+                self,
+                alpha=1.0,
+                fit_intercept=True,
+                normalize="deprecated" if sklearn_check_version("1.0") else False,
+                copy_X=True,
+                max_iter=None,
+                tol=1e-4,
+                solver="auto",
+                positive=False,
+                random_state=None,
+            ):
+                super().__init__(
+                    alpha=alpha,
+                    fit_intercept=fit_intercept,
+                    normalize=normalize,
+                    copy_X=copy_X,
+                    max_iter=max_iter,
+                    solver=solver,
+                    tol=tol,
+                    positive=positive,
+                    random_state=random_state,
+                )
+
+        def fit(self, X, y, sample_weight=None):
+            # It is necessary to properly update coefs for predict if we
+            # fallback to sklearn in dispatch
+            if hasattr(self, "_onedal_estimator"):
+                del self._onedal_estimator
+
+            dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_RidgeRegression.fit,
+                },
+                X,
+                y,
+                sample_weight,
+            )
+            return self
+
+        @wrap_output_data
+        def predict(self, X):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "predict",
+                {
+                    "onedal": self.__class__._onedal_predict,
+                    "sklearn": sklearn_RidgeRegression.predict,
+                },
+                X,
+            )
+
+        @wrap_output_data
+        def score(self, X, y, sample_weight=None):
+            check_is_fitted(self)
+
+            return dispatch(
+                self,
+                "score",
+                {
+                    "onedal": self.__class__._onedal_score,
+                    "sklearn": sklearn_RidgeRegression.score,
+                },
+                X,
+                y,
+                sample_weight=sample_weight,
+            )
+
+        def _test_type_and_finiteness(self, X_in):
+            xp, _ = get_namespace(X_in)
+            X = xp.asarray(X_in)
+
+            if np.iscomplexobj(X):
+                return False
+
+            try:
+                _assert_all_finite(X)
+            except BaseException:
```

Comment:
shouldn't be ValueError here?

Reply:
not sure about this one, took it from other algos. Since the function only returns True/False I would assume it does not matter what this except catches, any exception means the "test" failed but we really want the parent function test_type_and_finiteness to proceed and the "error" (False) to be handled within patching_status that wraps around it. Another reason might be that the implementation of _assert_all_finite function uses methods from numpy/sklearn, so to be safe against the exceptions those could raise BaseException is used. Overall, I agree that we lose some granularity of potential errors, so if you think the use of BaseException is still unjustifiable, we can agree on the change.

Reply:
outdated